presto-main/src/test/java/com/facebook/presto/security/TestFileBasedSystemAccessControl.java

```diff
@@ -0,0 +1,160 @@
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.security.Principal;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.facebook.presto.spi.security.Privilege.SELECT;
+import static com.facebook.presto.transaction.TransactionBuilder.transaction;
+import static com.facebook.presto.transaction.TransactionManager.createTestTransactionManager;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+public class TestFileBasedSystemAccessControl
+{
+    private static final Principal PRINCIPAL = new TestingPrincipal("principal");
+    private static final Identity alice = new Identity("alice", Optional.of(PRINCIPAL));
+    private static final Identity bob = new Identity("bob", Optional.of(PRINCIPAL));
+    private static final Identity admin = new Identity("admin", Optional.of(PRINCIPAL));
+    private static final Identity nonAsciiUser = new Identity("ÎÎÎ", Optional.of(PRINCIPAL));
+    private static final Set<String> allCatalogs = ImmutableSet.of("secret", "open-to-all", "all-allowed", "alice-catalog", "allowed-absent", "ÅÅÅ");
+    private static final QualifiedObjectName aliceTable = new QualifiedObjectName("alice-catalog", "schema", "table");
+    private static final QualifiedObjectName aliceView = new QualifiedObjectName("alice-catalog", "schema", "view");
+    private static final CatalogSchemaName aliceSchema = new CatalogSchemaName("alice-catalog", "schema");
+    private TransactionManager transactionManager;
+
+    @BeforeClass
+    public void setUp()
+    {
+        transactionManager = createTestTransactionManager();
+    }
+
+    @Test
+    public void testCatalogOperations()
+    {
+        AccessControlManager accessControlManager = new AccessControlManager(transactionManager);
+
+        accessControlManager.setSystemAccessControl(FileBasedSystemAccessControl.NAME, ImmutableMap.of(
+                "security.config-file", "src/test/resources/catalog.json"));
+
+        transaction(transactionManager, accessControlManager)
+                .execute(transactionId -> {
+                    assertEquals(accessControlManager.filterCatalogs(admin, allCatalogs), allCatalogs);
+                    Set<String> aliceCatalogs = ImmutableSet.of("open-to-all", "alice-catalog", "all-allowed");
+                    assertEquals(accessControlManager.filterCatalogs(alice, allCatalogs), aliceCatalogs);
+                    Set<String> bobCatalogs = ImmutableSet.of("open-to-all", "all-allowed");
+                    assertEquals(accessControlManager.filterCatalogs(bob, allCatalogs), bobCatalogs);
+                    Set<String> nonAsciiUserCatalogs = ImmutableSet.of("open-to-all", "all-allowed", "ÅÅÅ");
+                    assertEquals(accessControlManager.filterCatalogs(nonAsciiUser, nonAsciiUserCatalogs), nonAsciiUserCatalogs);
+                });
+    }
+
+    @Test
+    public void testSchemaOperations()
```

Comment:
Hopefully you'll be able to eliminate all of the below tests if you can get rid of the catalog check in the rest of the fields of the access control